vaas/vaas/cluster/models.py

```diff
@@ -105,32 +116,75 @@ def validate_unique(self, exclude=None):
                     )
 
 
-class DomainMapping(models.Model):
+class DomainMapping(models.Model, AbsModelWithJsonField):
     TYPE_CHOICES = (
         ('static', 'Static'),
         ('dynamic', 'Dynamic')
     )
     domain = models.CharField(max_length=128, unique=True)
-    mapping = models.CharField(max_length=128)
+    mappings_list = models.CharField(max_length=500, default='[]')
     type = models.CharField(max_length=7, choices=TYPE_CHOICES, default='static')
     clusters = models.ManyToManyField(LogicalCluster)
+    _mappings = None
+
+    @property
+    def mappings(self) -> Set[str]:
+        return self._get_parsed_field(self._mappings, self.mappings_list)
+
+    @mappings.setter
+    def labels(self, mappings: Union[List, Set]) -> None:
+        self._mappings, self.mappings_list = self._prepare_set_and_json(mappings)
 
     def mapped_domain(self, cluster: LogicalCluster) -> str:
-        result = str(self.mapping)
+        # TODO: should be transformed to mapped_domains -> List[str] in consecutive change
+        result = str(list(self.mappings)[0])
         if self.type == 'dynamic':
-            result = result.format(**self._prepare_placeholders(cluster.labels))
+            cluster_labels = set(list(cluster.parsed_labels().keys()))
+            for mapping, required_labels in self.__parse_placeholders().items():
+                if not required_labels.difference(cluster_labels):
+                    result = mapping.format(**cluster.parsed_labels())
         return result
 
-    def _prepare_placeholders(self, labels: Set[str]) -> Dict[str, str]:
+    def is_cluster_related_by_labels(self, cluster: LogicalCluster) -> bool:
+        result = False
+        cluster_labels = set(list(cluster.parsed_labels().keys()))
+        # check if all needed placeholders of any mapping are satisfied by cluster labels
+        for required_labels in self.__parse_placeholders().values():
+            result = result or not required_labels.difference(cluster_labels)
+        return result
+
+    def __prepare_placeholders_arguments(self, labels: Set[str]) -> Dict[str, str]:
```

Comment:
This is duplicated logic with:
```python
def parsed_labels(self) -> Dict[str, str]:
```
could move it to ABSClass?

Reply:
No, it was replaced by parsed_labels and it is not used anymore. I will clean that